autoscheduler/frontend/src/components/SchedulingPage/SchedulingPage.tsx

```diff
@@ -68,6 +99,17 @@ const SchedulingPage: React.FC<SchedulingPageProps> = ({
               {`Total Hours: ${hoursForSchedule(currentSchedule)}`}
             </Typography>
             <div className={styles.rightButtonContainer}>
+              {loadingScreenshot ? (
+                <div className={styles.loadingIndicatorContainer}>
+                  <SmallFastProgress />
+                </div>
+              ) : (
+                <Tooltip title="Save as image">
+                  <IconButton onClick={(): void => handleClick()}>
+                    <SaveAlt color="primary" />
+                  </IconButton>
+                </Tooltip>
+              )}
```

Comment:
Comment for this and the other ternary in the return value: I prefer moving conditional rendering outside of component returns to keep it somewhat organized